### ISSUE TITLE
Remove/clear extension

### ISSUE THREAD

**Issue by luixxiul** [brave/browser-laptop#6794]
Users should be able to permanently delete extensions from the browser

Currently, extensions are not installed until they are enabled. This is almost the same as being deleted (since it doesn't exist)... but it still shows in the extensions panel. Users should be able to remove it from there.

## Original issue description
**Describe the issue you encountered:**

https://community.brave.com/t/delete-remove-an-extension/807?u=suguru
> How do I Delete/Remove an Extension. Ex: I accidentally added Pocket and would like to remove it.

- Brave Version (revision SHA): 0.12.15

- Steps to reproduce:
    1. Open about:extensions
    
- Actual result:
The extensions cannot be removed

- Expected result:
They can be removed

- Any related issues:

**Comment by Liunkae** [comment]
The extensions (e.g. Pocket) be disabled in Preferences > Advanced…

**Comment by luixxiul** [comment]
I think the user wanted to delete it, not disabling it

**Comment by Liunkae** [comment]
Question: Should users be able to remove extensions that are bundled with Brave?

**Comment by luixxiul** [comment]
The extensions are fetched remotely not being bundled: https://github.com/brave/browser-laptop/issues/4081

**Comment by Liunkae** [comment]
Thanks @luixxiul, I misunderstood. But the extensions are still "forced" to be installed, right? (Or by default, I suppose.)

**Comment by Rasmis** [comment]
I chose Brave instead of Firefox because of Mozilla's decision to force third-party-privacy-invading-closed-source-safe-habour-violating spyware Pocket™ upon their innocent users. So I was saddened to see that it is also being pushed upon the users of Brave.

What is the thinking behind doing this?

**Comment by luixxiul** [comment]
It is not enabled by default

**Comment by Rasmis** [comment]
@luixxiul I know. But it's there. It is pushed upon the users. Like an ad for BigMac™ on the inside of your glasses. There's no burger, but the reminder is always there. It's not heavy, but it's not weightless.

**Comment by luixxiul** [comment]
> But it's there.

It will not be downloaded until you enable it. Please just ignore it, thanks

**Comment by luixxiul** [comment]
@Rasmis if you do not want to see even its name, please create a new issue to suggest something like "hide extensions name", thanks

**Comment by Liunkae** [comment]
I am fine with the existing behavior as long as extensions like Pocket:
1. Are not downloaded or enabled by default.
2. Are completely disabled when turned off.

**Comment by srirambv** [comment]
It can be implemented something like this 
![image](https://cloud.githubusercontent.com/assets/17010094/22475057/b70fade0-e7d5-11e6-8144-86ef932a511b.png)


![image](https://cloud.githubusercontent.com/assets/17010094/22475143/ef5dded8-e7d5-11e6-821f-8551aa5fea75.png)

User can enable which one they want and if extensions are turned off disable/delete the installed extensions

cc: @bradleyrichter

**Comment by bsclifton** [comment]
Fixed with https://github.com/brave/browser-laptop/pull/8065

**Comment by cezaraugusto** [comment]
Reopening after #8264

**Comment by CalAnon46** [comment]
I strongly support this. Brave should not be forcing us to accept anything we don't absolutely want - I don't care if it's "not preinstalled" or not, it should NOT BE SUGGESTED AT ALL. I don't WANT those weird extensions on my list, I never asked for them. I want to use BRAVE as a replacement for Firefox, but I can't justify that in its current state.

**Comment by jpliew** [comment]
Strongly support this too. I can't agree more with @CalAnon46 , we don't need those weird extensions on our list, installed or not.

We already have so many annoyances in Chrome and Firefox and are trying to move on to a new browser. If Brave wish to capture the new market, please, CLEAN and FAST will be the best selling point to take down Chrome and Firefox.

**Comment by jordanglassman** [comment]
This issue was opened asking about deleting/removing installed extensions, and then morphed into other topics, like pre-installed extensions and disabling installed extensions.

Disabling and pre-installed deserve their own issues, and the ability to fully remove installed extensions should be kept alive as a feature request.  Obviously a modern web browser should have this ability.

**Comment by jordanglassman** [comment]
Somehow extensions are persisted in global settings somewhere.  Even

```
$ rm -rf ~/Library/Application\ Support/brave/
$ rm -rf /Applications/Brave.app/
```

does not fully remove installed extensions from the list when Brave is reopened.  They still appear in the list, although they are disabled and specify version "Latest".  

However this workaround does result in previously installed extensions being deleted from the `~/Library/Application\ Support/brave/Extensions` directory.

**Comment by Jacalz** [comment]
+1 from me
![image](https://user-images.githubusercontent.com/25466657/34916439-c0b8ff8e-f938-11e7-83ef-2d7623a8410d.png)
We could just have an X for every extension and those x icons could light up if there are files of that extension on the computer 👍

**Comment by BeBlockTech** [comment]
My biggest issue is when we install metamask, you can't be deleted anymore!! I even uninstalled the browser, ran a cleaner, deleted the files in the roaming folder, and MetaMask is still there with my account linked to it. 

Can someone tell me how to reset this browser to its default?

**Comment by bsclifton** [comment]
@BeBlockTech you should be able to delete the extension's information by deleting the appropriate folder in your profile directory.  Your profile directory will be:
`%APPDATA/brave/` on Windows
`~/Library/Application Support/brave/` on macOS
`~/.config/brave/` on Linux

The extension info should be under the `Extensions` subdirectory. For example, I am on macOS... I can delete like so:
```
rm -rf ~/Library/Application Support/brave/Extensions/nkbihfbeogaeaoehlefnkodbefgpgknn/
```

**Comment by safecoms** [comment]
<img width="763" alt="screenshot 2018-04-21 13 42 49" src="https://user-images.githubusercontent.com/24518544/39081349-ec02e75e-4569-11e8-83a3-e2a1110e6170.png">

I did not enabled Dashline but it has 13MB of traffic out from Brave, I want to disable it, this is a password plugin that I am not using and it is talking to servers in London and Malaysia
WtF is this?

**Comment by safecoms** [comment]
this is on a Mac and the app is from littlesnitch
it is extremely useful and fairly priced.
I have had this for years, this is the latest version and it is extremely well designed

**Comment by EricCreatives** [comment]
" rm -rf ~/Library/Application Support/brave/Extensions/nkbihfbeogaeaoehlefnkodbefgpgknn " 
On terminal for macOS doesn't do the trick. 

When I enable Metamask again I'm still prompted to login to my previous Metamask account. 
I'd like to be able to setup a new wallet / account with Metamask. Kindly advice.

**Comment by mohsenghajar** [comment]
Same here, need to start fresh with Metamask.  Brave does not like this and still wants me to login to the existing one.  No option to start new account.

**Comment by kdawg1406** [comment]
Makes no sense, not being able to uninstall any extension from my computer.  Please add this feature.

**Comment by AWulkan** [comment]
Seriously, why are these third-party extensions even suggested in the browser? I don't want my list filled up with random extensions that I will never use. Who needs 4 different password managers for example? Also, does Brave take personal responsibility if one of them has vulnerabilities?

The reasoning that must have taken place behind the decicion to not make it easy to remove them is mind boggling.

**Comment by Jacalz** [comment]
Just to shed some light on this issue since most people don’t seem to know that the extensions aren’t installed until you enable them. (Not just here but in general) So it’s not like all extensions are installed, that is not what his issue is about... This issue is about not being able to remove the extensions that have been installed by the user!

**Comment by bsclifton** [comment]
Closing as `wontfix` as we won't be able to address with our current release

[brave-core](https://github.com/brave/brave-browser) DOES let you remove all extensions though, which addresses this issue. It's coming soon- stay tuned

**Comment by jpliew** [comment]
Great!! after 1.5 year finally we know the answer.

**Comment by luishrd** [comment]
I guess, the promise of privacy was yet another lie. Time to uninstall

**Comment by bsclifton** [comment]
@luishrd if you install the new version of brave (https://brave.com/download), you can uninstall the extensions. This repo is tracking the older Brave browser which did not let you do so.

Uninstalling the old version is a good idea actually- please do and install the new one. You can use the importer to pull in all your bookmarks, etc.
